src/wxflow/task.py

```diff
@@ -91,3 +96,17 @@ def clean(self):
         Methods to clean after execution and finalization prior to closing out a task
         """
         pass
+
+    def extend_task_config(self, local_dict: Union[Dict, AttrDict]) -> None:
+        """
+        Extend task_config attribute-dictionary with another dictionary
+        """
+
+        self.task_config = AttrDict(**self.task_config, **local_dict)
```

Comment:
This breaks provenance.
The intention is to create a task_config in the `init`.  By allowing this, users can add/edit/remove items from task_config and it gets increasingly difficult to find out where what was added/removed.
As much as possible, we should extract only required pieces of information.  If we turn these into `global` variables, it can get complicated quickly.

Reply:
Users are already doing this when they create task_config in places like the atm_analysis.py (https://github.com/NOAA-EMC/global-workflow/blob/205d0c2b13e2d7755cec75bf8c978ab20d453862/ush/python/pygfs/task/atm_analysis.py#L59) where `config`, `runtime_config`, and `local_dict` are combined into one task config. Even if I remove this method, there's nothing stopping someone from modifying `task_config`. 

One possibility is to make `task_config` also private but accessible (read-only) using a public method, something like `mytask.get_task_config()['foo']`. In the case of the atm_analysis.py subclass `__init__`, one can just initialize it with like so:

`super().__init__( AttrDict( **config, **local_dict ) )`

and instead just create `local_dict` first before calling the above. Thus the user has one shot to add to `task_config` before it becomes read-only.